source-mongodb/main_test.go

```diff
@@ -806,3 +806,62 @@ func (v *correctnessInvariantsCaptureValidator) Summarize(w io.Writer) error {
 func (v *correctnessInvariantsCaptureValidator) bindingIdentifier(database, collection string) string {
 	return database + "." + collection
 }
+
+// TestNestedCursorFields tests nested cursor field lookups like "a.b".
+func TestNestedCursorFields(t *testing.T) {
+	const (
+		database   = "testDb"
+		collection = "nestedCursorCollection"
+	)
+	ctx := context.Background()
+	client, cfg := testClient(t)
+
+	cleanup := func() {
+		require.NoError(t, client.Database(database).Drop(ctx))
+	}
+	cleanup()
+	t.Cleanup(cleanup)
+
+	binding := makeBinding(t, database, collection, captureModeIncremental, "a.b")
+	cs := &st.CaptureSpec{
+		Driver:       &driver{},
+		EndpointSpec: &cfg,
+		Checkpoint:   []byte("{}"),
+		Validator:    &st.OrderedCaptureValidator{},
+		Sanitizers:   commonSanitizers(),
+		Bindings:     []*flow.CaptureSpec_Binding{binding},
+	}
+
+	col := client.Database(database).Collection(collection)
+	refTime := time.Date(2023, 11, 7, 16, 57, 0, 0, time.UTC)
+
+	advanceCapture(ctx, t, cs)
+
+	for idx := 0; idx < 5; idx++ {
+		doc := map[string]any{
+			"a": map[string]any{
+				"b": refTime.Add(time.Duration(idx) * time.Second),
+			},
+			"data": fmt.Sprintf("test data %d", idx),
+		}
+		_, err := col.InsertOne(ctx, doc)
+		require.NoError(t, err)
+	}
+
+	backdateState(t, cs, boilerplate.StateKey(binding.StateKey))
+	advanceCapture(ctx, t, cs)
+
+	for idx := 5; idx < 10; idx++ {
+		doc := map[string]any{
+			"a": map[string]any{
+				"b": refTime.Add(time.Duration(idx) * time.Second),
+			},
+			"data": fmt.Sprintf("test data %d", idx),
+		}
+		_, err := col.InsertOne(ctx, doc)
+		require.NoError(t, err)
+	}
+
+	backdateState(t, cs, boilerplate.StateKey(binding.StateKey))
+	advanceCapture(ctx, t, cs)
```

Comment:
Need to actually do a snapshot, and that should be committed as well. See https://github.com/estuary/connectors/blob/32bd7d5f32802da101278046a3d015d00de91738/source-mongodb/main_test.go#L267